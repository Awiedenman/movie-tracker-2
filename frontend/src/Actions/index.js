export const fetchInitialMovies = movies => ({
  type: 'FETCH_INITIAL_MOVIES',
  movies
});

export const userLogin = user => ({
  type: 'USER_LOGIN',
  id: user.id,
  name: user.name,
  loggedIn: user.loggedIn
});

export const userSignUp = newUserData => ({
  type: 'USER_SIGN_UP',
  id: newUserData.id
});

export const userSignOut = () => ({
  type: 'USER_SIGN_OUT'
});

export const clearUserFavorites = () => ({
  type: 'CLEAR_USER_FAVORITES'
});

export const addFavorite = movie => ({
  type: 'ADD_FAVORITE',
  favorite: movie
});

export const removeFavorite = () => ({
  type: 'REMOVE_FAVORITE'
});

export const addExistingFavorites = existingUserFavorites => {
  return {
    type: 'ADD_EXISTING_FAVORITES',
    favorites : existingUserFavorites
  };
};