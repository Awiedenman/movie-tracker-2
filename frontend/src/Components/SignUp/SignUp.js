import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export class SignUp extends Component {
  constructor( props ){
    super( props );
    this.state = {
      email:'',
      password: ''
    };
  }

  handleChange=(event)=> {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  }

  handleSubmit=(event)=> {
    event.preventDefault();
    
  }

  render() {
    return (
      <section>
        <h1>Sign Up</h1>
        <form onSubmit={this.handleSubmit}>
          <input onChange={this.handleChange} type="text" placeholder="email" name='email' value={this.state.email}/>
          <input onChange={this.handleChange} type="password" placeholder="password" name='password' value={this.state.value}/>
          <button>Sign Up</button>
        </form>
      </section>
    );
  }
}

export const mapStateToProps = (state) => {
  userInfo: state.userInfo
};

export const mapDispatchToProps = () => {

};

export default connect(mapStateToProps, mapDispatchToProps)(SignUp);

SignUp.propTypes = {
};
