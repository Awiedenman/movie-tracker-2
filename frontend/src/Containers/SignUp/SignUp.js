import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { userSignUpRequest } from '../../helpers/apiCalls';
import * as actions from '../../Actions';

export class SignUp extends Component {
  constructor( props ){
    super( props );
    this.state = {
      userName: '',
      email:'',
      password: '',
      failedSignUp: false,
      error: null
    };
  }

  handleChange=(event)=> {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  }

  handleSubmit= async (event)=> {
    event.preventDefault();
    const { userName, email, password} = this.state;
    try {
      const newUserData = await userSignUpRequest( userName, email, password);
      this.props.handleUserSignUp(newUserData);
    } catch (error) {
      this.setState({ error, failedSignUp: true })
    }
    this.setState({userName: '', email: '', password: ''});
  }

  render() {
    return (
      <section>
        <h1>Sign Up</h1>
        { this.state.failedSignUp && <p>{this.state.error.message}</p>}
        <form onSubmit={this.handleSubmit}>
          <input onChange={this.handleChange} type="text" placeholder="name" name="userName" value={this.state.userName}/>
          <input onChange={this.handleChange} type="text" placeholder="email" name='email' value={this.state.email}/>
          <input onChange={this.handleChange} type="password" placeholder="password" name='password' value={this.state.password}/>
          <button>Sign Up</button>
        </form>
      </section>
    );
  }
}

export const mapStateToProps = (state) => ({
  userInfo: state.userInfo
});

export const mapDispatchToProps = (dispatch) => ({
  handleUserSignUp: (newUserData) => dispatch(actions.userSignUp(newUserData))
});

export default connect(mapStateToProps, mapDispatchToProps)(SignUp);

SignUp.propTypes = {
  handleUserSignUp: PropTypes.func
};
