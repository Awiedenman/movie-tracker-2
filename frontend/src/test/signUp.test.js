import React from 'react';
import { shallow } from 'enzyme';
import { SignUp, mapDispatchToProps } from '../Containers/SignUp/SignUp';
import { userSignUp } from '../Actions';


describe('SignUp', () => {
  let wrapper;

  beforeEach(() => wrapper = shallow( < SignUp handleUserSignUp={ jest.fn() }/>));

  describe('handleChange', () => {

    test('should update the name prop in state on change ', () => {
      const mockEvent= {
        target:{
          name:'name',
          value: 'Austin'
        }
      };

      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state('name')).toEqual(mockEvent.target.value);
    });

    test('should update prop email in state onChange', () => {
      const mockEvent = {
        target :{
          name: "email",
          value: "austin@aol.com"
        }
      };
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state('email')).toEqual(mockEvent.target.value);
    });

    test('should update password prop in state onChange', () => {
      const mockEvent = {
        target: {
          name: 'password',
          value: 'password'
        }
      };
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state('password')).toEqual(mockEvent.target.value);
    });
  });

  describe('handleSubmit', () => {
    test('should call handleUserInput with the correct arguments', async () => {
      const mockResponse = ({ id: 34, status: 'success' });
      const mockHandleUserSignUp = jest.fn();
      const mockEvent = { preventDefault: jest.fn() };

      window.fetch = jest.fn().mockImplementation(() =>
        Promise.resolve({
          ok: true,
          json: () => Promise.resolve(mockResponse)
        }));

      wrapper = shallow(
        <SignUp
          handleUserSignUp={mockHandleUserSignUp}
        />
      );

      await wrapper.instance().handleSubmit(mockEvent);
      expect(mockHandleUserSignUp).toHaveBeenCalledWith(mockResponse);
    });

    test('should reset the value of state to an empty string on submit', async () => {
      const mockEvent = {
        preventDefault: jest.fn()
      };

      wrapper.setState({
        name: 'Austin',
        email: 'austin@aol.com',
        password: 'password'
      });

      await wrapper.instance().handleSubmit(mockEvent);
      expect(wrapper.state('userName')).toEqual('');
      expect(wrapper.state('email')).toEqual('');
      expect(wrapper.state('password')).toEqual('');
    });

    test('should throw error when signUp fails', async () => {
      const mockEvent = {
        preventDefault: jest.fn()
      };
      window.fetch = jest.fn().mockImplementation(() =>
        Promise.reject({
          ok: false
        }));

      await wrapper.instance().handleSubmit(mockEvent);
      expect(window.fetch).toHaveBeenCalled();
      expect(wrapper.state('failedSignUp')).toEqual(true);
    });
  });

  describe('mapDispatchToProps', () => {
    test('should call dispatch with a userSignUp action when handleUserSignUp is called', () => {
      const mockDispatch = jest.fn();
      const mockActionToDispatch = userSignUp('Austin', 'austin@aol.com', 'password');

      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.handleUserSignUp('Austin', 'austin@aol.com', 'password');

      expect(mockDispatch).toHaveBeenCalledWith(mockActionToDispatch);
    });
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});