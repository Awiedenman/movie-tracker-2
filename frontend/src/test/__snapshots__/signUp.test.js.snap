// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp
    handleUserSignUp={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="sign-up__title"
>
          Sign Up
</h1>,
        false,
        <form
          className="form"
          onSubmit={[Function]}
>
          <input
                    className="form__input"
                    name="userName"
                    onChange={[Function]}
                    placeholder="Name"
                    required={true}
                    type="text"
                    value=""
          />
          <input
                    className="form__input"
                    name="email"
                    onChange={[Function]}
                    placeholder="Email"
                    required={true}
                    type="text"
                    value=""
          />
          <input
                    className="form__input"
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    required={true}
                    type="password"
                    value=""
          />
          <button
                    className="form__button"
          >
                    Sign Up
          </button>
</form>,
      ],
      "className": "sign-up",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign Up",
          "className": "sign-up__title",
        },
        "ref": null,
        "rendered": "Sign Up",
        "type": "h1",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="form__input"
              name="userName"
              onChange={[Function]}
              placeholder="Name"
              required={true}
              type="text"
              value=""
/>,
            <input
              className="form__input"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              required={true}
              type="text"
              value=""
/>,
            <input
              className="form__input"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              required={true}
              type="password"
              value=""
/>,
            <button
              className="form__button"
>
              Sign Up
</button>,
          ],
          "className": "form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form__input",
              "name": "userName",
              "onChange": [Function],
              "placeholder": "Name",
              "required": true,
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form__input",
              "name": "email",
              "onChange": [Function],
              "placeholder": "Email",
              "required": true,
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form__input",
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "required": true,
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up",
              "className": "form__button",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="sign-up__title"
>
            Sign Up
</h1>,
          false,
          <form
            className="form"
            onSubmit={[Function]}
>
            <input
                        className="form__input"
                        name="userName"
                        onChange={[Function]}
                        placeholder="Name"
                        required={true}
                        type="text"
                        value=""
            />
            <input
                        className="form__input"
                        name="email"
                        onChange={[Function]}
                        placeholder="Email"
                        required={true}
                        type="text"
                        value=""
            />
            <input
                        className="form__input"
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        required={true}
                        type="password"
                        value=""
            />
            <button
                        className="form__button"
            >
                        Sign Up
            </button>
</form>,
        ],
        "className": "sign-up",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
            "className": "sign-up__title",
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "h1",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="form__input"
                name="userName"
                onChange={[Function]}
                placeholder="Name"
                required={true}
                type="text"
                value=""
/>,
              <input
                className="form__input"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                required={true}
                type="text"
                value=""
/>,
              <input
                className="form__input"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                required={true}
                type="password"
                value=""
/>,
              <button
                className="form__button"
>
                Sign Up
</button>,
            ],
            "className": "form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form__input",
                "name": "userName",
                "onChange": [Function],
                "placeholder": "Name",
                "required": true,
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form__input",
                "name": "email",
                "onChange": [Function],
                "placeholder": "Email",
                "required": true,
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form__input",
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "required": true,
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Up",
                "className": "form__button",
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
